test/functional/item-checkout.js

```diff
@@ -0,0 +1,151 @@
+import { Application } from 'spectron';
+import electron from 'electron-prebuilt';
+import { assert } from 'chai';
+import MockServer from '../util/mock-server';
+
+describe('item checkout', function () {
+
+    this.timeout(10000);
+    let app;
+    let mockServer = new MockServer();
+
+    before(() => {
+        app = new Application({
+            path: electron,
+            args: ['index.js', '--port=8080']
+        });
+        return app.start().then(() => {
+            return mockServer.start(8080);
+        });
+    });
+
+    after(() => {
+        if (app && app.isRunning()) {
+            return app.stop().then(() => {
+                return mockServer.stop();
+            });
+        }
+    });
+
+    beforeEach(() => {
+        mockServer.clearExpectations();
+    });
+
+    it('loads the homepage', () => {
```

Comment:
Not sure if this test should be a part of the check-in functional tests. but it's not that big of a deal.